chia/solver/solver_api.py

```diff
@@ -40,12 +40,12 @@ async def solve(
             self.log.error("Solver is not started")
             return None
 
-        self.log.debug(f"Solving partial {request.partial_proof.hex()}")
+        self.log.debug(f"Solving partial {request.partial_proof[:5]}")
 
         try:
-            proof = self.solver.solve(request.partial_proof)
+            proof = self.solver.solve(request.partial_proof, request.plot_id, request.strength, request.size)
```

Comment:
do we need the plot_id  and size for solving ? (i couldn't find any mention of this in the channel)

you can know the size from the partial iirc

i have vague memory of us talking about encoding the strength in the partial did we decide not to ? (i prefer separating it from the partial as well)

Reply:
we definitely need the plot ID. It's used for the `ProofParams` here: https://github.com/Chia-Network/pos2-chip/blob/main/src/api.cpp#L98
Looking over this again, I'm not sure we can deduce *k* from `partial_proof`. I think `partial proof is always 64 integers. Each integer is at most `k` bits though, but I don't think that's enough. This is the C-function:
https://github.com/Chia-Network/pos2-chip/blob/main/src/api.cpp#L91-L116

This is the rust function:
https://github.com/Chia-Network/pos2-chip/blob/main/crates/chia-pos2/src/lib.rs#L107-L126